lesson1/src/arrays/arrayProblems.js

```diff
@@ -22,25 +25,52 @@ class ArrayProblems {
    */
   static getMax(array) {
     // todo: do stuff here
+    return array.reduce((acc, curr) => {
+      return acc > curr ? acc : curr;
+    })
   }
 
   /*
    Can you implement getMax again without using any js collection method?
    */
   static getMax2(array) {
     // todo: 🙌 do magic !
+    let result;
+    let i = 0;
+    let j = 0;
+    while(i < array.length){
+      while(j < array.length -1){
+        if(array[i] > array[j]){
+          result = array[i];
+        }else if(array[j] > array[i]){
+          result = array[j];
+        }
+        j++
+      }
+      i++
+      j = i;
+    }
+    // for (let i = 0; i < array.length; i++) {
+    //   for (let j = i; j < array.length - 1; j++) {
+    //     // todo: do stuff here
+    //     if (array[j] > array[i]) {
+    //       result = array[j];
+    //     } else if (array[i] > array[j]) {
+    //       result = array[i];
+    //     }
+    //   }
+    // }
+    return result;
   }
 
-  /*
-   A left rotation operation on an array shifts each
-   of the array's elements 1 unit to the left.
-   For example, if 2 left rotations are performed on array [1,2,3,4,5],
-   then the array would become [3,4,5,1,2].
-
-   This method should perform n left rotations on the array
-   */
   static rotateLeft(array, n){
     // todo: 🙌 do magic !
+    let count= 0;
+    while(count < n){
+      array.push(array.shift());
```

Comment:
Super easy: the Array.prototype.shift() if it is an array then it returns undefined, so one way of avoiding it to push undefined is : 💯 😄 
``` let count= 0;
    while(count < n){
      let item =array.shift();
      item ? array.push(item) : array.push([]);
      count++
    }
```